src/v/storage/segment_appender.cc

```diff
@@ -477,7 +477,6 @@ void segment_appender::dispatch_background_head_write() {
      * out) synchronously.
      */
     auto pending_write_sem = _pending_head_write;
-    auto units = ss::get_units(*pending_write_sem, 1);
```

Comment:
based on this comment above, it sounds the queuing here is important--that it happens uninterrupted (which it won't if it taken only after the concurrent flushes semaphore is acquired).

Reply:
It is important, without the ordering provided by the synchronous get_units, two appends which resolve in one order could have their writes resolve in a different order, breaking expectations in the code below (e.g., that the `full` write happens after any partial write and that the committed offset advances in increasing order).

Reply:
seems like we should close this PR given the other PR that is approved and getting ready to merge?